hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/utils/Pipelines.java

```diff
@@ -137,7 +136,7 @@ public static DataStreamSink<Object> bulkInsert(Configuration conf, RowType rowT
         // use #partitionCustom instead of #keyBy to avoid duplicate sort operations,
         // see BatchExecutionUtils#applyBatchExecutionSettings for details.
         Partitioner<String> partitioner = (key, channels) ->
-            KeyGroupRangeAssignment.assignKeyToParallelOperator(key, StreamGraphGenerator.DEFAULT_LOWER_BOUND_MAX_PARALLELISM, channels);
+            KeyGroupRangeAssignment.assignKeyToParallelOperator(key, KeyGroupRangeAssignment.UPPER_BOUND_MAX_PARALLELISM, channels);
         dataStream = dataStream.partitionCustom(partitioner, rowDataKeyGen::getPartitionPath);
```

Comment:
Does `KeyGroupRangeAssignment#computeDefaultMaxParallelism` work here ?

Reply:
It seems `KeyGroupRangeAssignment.computeDefaultMaxParallelism(conf.getInteger(FlinkOptions.WRITE_TASKS))` would be a possible solution?

Reply:
Could I use `conf.getInteger(FlinkOptions.WRITE_TASKS)` directly?
```java
// shuffle by partition keys
// use #partitionCustom instead of #keyBy to avoid duplicate sort operations,
// see BatchExecutionUtils#applyBatchExecutionSettings for details.
Partitioner<String> partitioner = (key, channels) -> KeyGroupRangeAssignment.assignKeyToParallelOperator(key,
                conf.getInteger(FlinkOptions.WRITE_TASKS), channels);
```

Reply:
No, we should use a max parallelism here.

Reply:
OK, I've pushed another:)